app/controllers/hub/organizations_controller.rb

```diff
@@ -28,9 +28,18 @@ def show
     end
 
     def index
-      @coalitions = Coalition.accessible_by(current_ability).includes(:organizations)
-      @organizations = @vita_partners.organizations.includes(:organization_capacity)
-      @independent_organizations = @organizations.where(coalition: nil)
+      # Load organizations slowly first, to avoid lots of queries later
+      organizations = @vita_partners.organizations.includes(:coalition, :child_sites, :organization_capacity).load
+
+      @organizations_by_coalition = if can? :read, Coalition
+         organizations.group_by(&:coalition).sort do |coalition_a, coalition_b|
```

Comment:
This sort is a little bit messy, but we have to deal with the fact that there is a nil in the bunch.

Reply:
For purposes of the view and looping through the data one time, we set up every role to loop through @organizations_by_coalition, but some roles don't have access to Coalitions so we force a nil coalition.

Reply:
I encourage an approach where you separate the nil values from the non-nil values, so that you can treat them differently.

I also personally enjoy the 'guard clause' style, e.g.

```
return 1 if coalition_a[0].nil? # independent orgs at the end

coalition_a[0].name <=> coalition_b[0].name
```

The approach you've taken here is fine, too.

Reply:
Oh that's way better.